src/app/components/homepage/recent-user-actions/recent-user-actions.component.html

```diff
@@ -0,0 +1,18 @@
+<div class="card">
+  <div class="card-header">
+    <h3>Recent Actions</h3>
+  </div>
+  <ul class="list-group list-group-flush">
+    <ng-container *ngIf="userActionsHtml; else noRecentActions">
+      <ng-container *ngFor="let action of userActionsHtml.slice(0,5)">
```

Comment:
All the logic should be in the TS component, not the HTML. please move the slice in the TS.